azurerm/internal/services/loganalytics/log_analytics_storage_insights_resource.go

```diff
@@ -0,0 +1,211 @@
+package loganalytics
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/operationalinsights/mgmt/2020-03-01-preview/operationalinsights"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/loganalytics/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/loganalytics/validate"
+	storageValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmLogAnalyticsStorageInsights() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLogAnalyticsStorageInsightsCreateUpdate,
+		Read:   resourceArmLogAnalyticsStorageInsightsRead,
+		Update: resourceArmLogAnalyticsStorageInsightsCreateUpdate,
+		Delete: resourceArmLogAnalyticsStorageInsightsDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.LogAnalyticsStorageInsightsID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.LogAnalyticsStorageInsightsName,
+			},
+
+			// must ignore case since API lowercases all returned data
+			// IssueP https://github.com/Azure/azure-sdk-for-go/issues/13268
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
```

Comment:
this can become case sensitive since we're defining the Resource ID rather than relying on the one coming back from the API here:

```suggestion
			"resource_group_name": azure.SchemaResourceGroupName(),
```

(and remove the comments)

Reply:
actually can't we parse this from the Workspace ID?

Reply:
They don't need to be the same RG afaik?